src/PowerShellEditorServices/Language/TokenOperations.cs

```diff
@@ -28,29 +29,27 @@ internal static class TokenOperations
         {
             List<FoldingReference> foldableRegions = new List<FoldingReference>();
 
-            // Find matching braces { -> }
-            foldableRegions.AddRange(
-                MatchTokenElements(tokens, TokenKind.LCurly, TokenKind.RCurly, RegionKindNone)
-            );
-
-            // Find matching braces ( -> )
+            // Find matching braces  { -> }
+            // Find matching hashes @{ -> }
             foldableRegions.AddRange(
-                MatchTokenElements(tokens, TokenKind.LParen, TokenKind.RParen, RegionKindNone)
+                MatchTokenElements(tokens, new TokenKind[] { TokenKind.LCurly, TokenKind.AtCurly }, TokenKind.RCurly, RegionKindNone)
```

Comment:
Could this array be created in a static readonly field?

Reply:
Yeah it could be

Reply:
Given how often we'll be folding, it's probably a good idea -- want to avoid GC pressure if we can. Should just be a matter of putting the following at the top of the class:

```csharp
// Opening tokens for { } and @{ }
private static readonly TokenKind[] s_openingBraces = new [] { TokenKind.LCurly, TokenKind.AtCurly };

// Opening tokens for ( ), @( ) and $( )
private static readonly TokenKind[] s_openingParens = new [] { TokenKind.LParen, TokenKind.AtParen, TokenKind.DollarParen };
```

Reply:
Agreed. This codepath is removed in PR #806 and we use the AST